sdk/eventhubs/azure_messaging_eventhubs/src/models/event_data.rs

```diff
@@ -0,0 +1,522 @@
+///    .with_correlation_id("correlation_id")
+///    .with_message_id("message_id")
+///    .add_property("key".to_string(), "value")
+///    .build();
+///
+/// println!("{:?}", event_data);
+/// ```
+///
+#[derive(Default, PartialEq, Clone)]
+pub struct EventData {
+    body: Option<Vec<u8>>,
+    content_type: Option<String>,
+    correlation_id: Option<MessageId>,
+    message_id: Option<MessageId>,
+    properties: Option<HashMap<String, AmqpValue>>,
+}
+
+impl EventData {
+    /// Creates a new builder to build an `EventData`.
+    pub fn builder() -> builders::EventDataBuilder {
+        builders::EventDataBuilder::new()
+    }
+
+    /// The properties of the event.
+    pub fn properties(&self) -> Option<&HashMap<String, AmqpValue>> {
+        self.properties.as_ref()
+    }
+
+    /// The body of the event.
+    pub fn body(&self) -> Option<&[u8]> {
+        self.body.as_deref()
+    }
+
+    /// The content type of the event, if one was specified.
+    pub fn content_type(&self) -> Option<&str> {
+        self.content_type.as_deref()
+    }
+
+    /// The correlation identifier of the event, if one was specified.
+    pub fn correlation_id(&self) -> Option<&MessageId> {
+        self.correlation_id.as_ref()
+    }
+
+    /// The message identifier of the event, if one was specified.
+    pub fn message_id(&self) -> Option<&MessageId> {
+        self.message_id.as_ref()
+    }
+
+    /// Convert the provided AMQP message into an EventData object.
+    fn from_message(message: &AmqpMessage) -> Self {
+        // // Create an EventData from the message.
+        let mut event_data_builder = EventData::builder();
+
+        // If the AMQP message body is a single binary value, copy it to
+        // the event data body.
+        if let AmqpMessageBody::Binary(binary) = message.body() {
+            if binary.len() == 1 {
+                event_data_builder = event_data_builder.with_body(binary[0].clone());
+            }
+        }
+
+        if let Some(properties) = message.properties() {
+            if let Some(content_type) = &properties.content_type {
+                event_data_builder = event_data_builder
+                    .with_content_type(Into::<String>::into(content_type.clone()));
+            }
+            if let Some(correlation_id) = &properties.correlation_id {
+                event_data_builder = event_data_builder.with_correlation_id(correlation_id.clone());
+            }
+            if let Some(message_id) = &properties.message_id {
+                event_data_builder = event_data_builder.with_message_id(message_id.clone());
+            }
+        }
+        if let Some(application_properties) = message.application_properties() {
+            for (key, value) in application_properties.0.clone() {
+                event_data_builder = event_data_builder.add_property(key, value);
+            }
+        }
+        event_data_builder.build()
+    }
+}
+
+impl Debug for EventData {
```

Comment:
Why not just derive `Debug`? Seems you've done nothing it wouldn't have.

Reply:
At one point there was a significant concern that because we don't have any idea what is in the body() of a message, it could be considered PII and we want to keep PII out of traces.

However, for now it's likely ok.